conftest.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+from enum import Enum
+from supplier import models
+
+
+class TestFields(Enum):
```

Comment:
`TestFields` is a very generic name, and you're putting it inside conftest.py at the root of your project, which means it will be visible to all tests in your repo. Is there a concrete reason to have it defined at this level? I'd suggest defining it with your tests, and if you find that you need it later in other places as well, then move it to a location relevant to everywhere you need it (i.e. make stuff directly visible at the level that you know it's needed, and not more).

I also have some reservations about creating a class here that you never actually instantiate. If you just want a list of names, then define a list of names and make them self explanatory to what they were created for.

i.e. something like this:
```
@pytest.fixture
def supplier_test_fields():
  return ["supplier_user_name", "supplier_first_name", "supplier_last_name", ...]
```

Lastly, you're test is missing some coverage for fizzy bubblech and muchentuchen.

Reply:
i use the tests fields in the conftest module and in supplier tests module, originally i made it in the root project to use it for delivery app.
as i see it there are three options:
- i can change its name to `SupTestFields` to make it more understandable.
- i can move it to supplier tests and import it to conftest, thou what was gained here? 
the purpose was that the rest of the programmers could use that code, isnt conftest module purpose 
is to create general code for the use of all?
- i could delete it and rewrite on both conftest and supplier tests, but isnt it code duplicating?

tell me which option you think is best.
*planning to do a whole module for hummus tests, those you mention will be added there.

Reply:
My preference would be to have it as a fixture (so it doesn't have to be imported) exposed to where it's required with a self-explanatory name that will not conflict with other fixture/variable names.

Reply:
well from what i understood conftest module should be used for all the tests in the code, correct me if im wrong, since that, all the fixture instances should be there as well, to create an instance ive made it easier and more readable (for me atleast)
by creating the enum class, so shouldnt it stay there (after name changed)?
trying to understand the conventions, dont mind changing it of coures.

Reply:
You can have conftest files on multiple levels.

In the example below, on the animals level you'd have fixtures relevant to all animals, and on the birds level you'd have fixtures relevant to all birds and so on.

```
→ tree .
.
└── animals
    ├── birds
    │   ├── conftest.py
    │   └── test_sparrow.py
    ├── conftest.py
    └── mammals
        ├── conftest.py
        └── test_dog.py
```

With that in mind, if you think that this is the right scope to have it defined, then I'm fine with it.

Reply:
i honestly dont mind, we can make folder for all the models together, your call.

Reply:
The idea is not to introduce more variables to a given namespaces than needed (a folder with everything will do the exact opposite). Those kind of things tend to surprise you when least expected. You find yourself trying to figure out why something has a value that you never dreamed on assigning to it, and 3 hours later realize there's a fixture somewhere with the exact same name that you were not aware about.

You're writing those tests and you know where you need the fixtures. I'm not going to make that call for you. My aim is just to have you aware of potential caveats.